packages/cozy-client/src/StackLink.js

```diff
@@ -77,11 +80,26 @@ export default class StackLink extends CozyLink {
     this.stackClient = null
   }
 
-  request(operation, result, forward) {
-    if (operation.mutationType) {
-      return this.executeMutation(operation, result, forward)
+  async request(operation, result, forward) {
+    if (this.isOnline && !(await this.isOnline())) {
+      return forward(operation)
     }
-    return this.executeQuery(operation)
+
+    try {
+      if (operation.mutationType) {
+        return await this.executeMutation(operation, result, forward)
+      }
+      return await this.executeQuery(operation)
+    } catch (err) {
+      if (isReactNativeOfflineError(err)) {
+        return forward(operation)
```

Comment:
I assume this is a fallback in case the `this.isOnline()` was not defined or wrongly returned true? I see the value, but I'm wondering whether it is good to have this kind of logic here. 
What if someone has a Cozy react-native app with only the StackLink defined? Here, it will forward the operation and the app will never receive the error

Reply:
> I assume this is a fallback in case the this.isOnline() was not defined or wrongly returned true?

Yes, this is mainly because when a device loose the Internet connection, it can take a few seconds before the offline status is triggered. This often happens when you are connected to an Internet box through Wifi. If you unplug the internet connection, then your computer/phones will still displayed the "online" icon for a few seconds.

> What if someone has a Cozy react-native app with only the StackLink defined? Here, it will forward the operation and the app will never receive the error

Good catch. But I'm not sure that we can check if any `next link` exist. I'll investigate how to do this correctly.